internal/remote/user_queue.go

```diff
@@ -3,109 +3,152 @@ package remote
 import (
 	"context"
 	"errors"
-
+	"github.com/ProtonMail/gluon/internal/queue"
 	"github.com/sirupsen/logrus"
 )
 
 var ErrQueueClosed = errors.New("the queue is closed")
 
-// process repeatedly pulls items off the operation queue and executes them.
-// TODO: What should we do with operations that failed to execute due to auth reasons?
-// We might want to save them somewhere so we can try again after the user has logged back in.
-func (user *User) process() {
-	defer user.processWG.Done()
+type userOpQueue struct {
+	poppedOps []operation
+	queue     *queue.CTQueue[operation]
+}
+
+func newUserOpQueue(queue *queue.CTQueue[operation]) userOpQueue {
```

Comment:
Why pass the queue in here DI-style? Any reason not to just construct a new queue internally?

Reply:
Good point. It was a left over from a previous state where that wasn't possible. Fixed.